bitcoin/src/pow.rs

```diff
@@ -1099,8 +1099,12 @@ impl kani::Arbitrary for U256 {
 mod tests {
     use super::*;
 
-    impl<T: Into<u128>> From<T> for Target {
-        fn from(x: T) -> Self { Self(U256::from(x)) }
+    impl From<u64> for Target {
```

Comment:
Why is this in tests in the first place?!

Reply:
Yeah, lol, I had the same thought. They're "mostly harmless" in that it's pretty hard for anyone downstream to compile the lib with `cfg(test)` on, but this is a super weird construction.

Reply:
I don't remember but I guess we did not implement `From`s for `Target` because such small targets are not that useful on mainnet?

Reply:
I think we should just call methods.

Reply:
> Why is this in tests in the first place?!

ooo you meant why are these even implemented, I thought you meant why are they not implemented outside of tests.

I can remove them and change the usage.

Reply:
You and I were writing at the same time @apoelstra I missed your comment til now.

Reply:
I kinda meant both, although `Target` is not a general-purpose number so I'd prefer not to have `From<{integer}>` for it.

Reply:
Removing them makes the tests worse, I tried using `from_hex` instead. The other option is using `from_be_bytes` which is even worse. Do you object strongly to the `From` impls done in test code Kix?

Reply:
> The other option is using `from_be_bytes` which is even worse.

Why is it worse? Long term I'd prefer not to do weird things like that but it's not an urgent issue.

Reply:
Added #3160